node_modules/oae-principals/lib/definitiveDeletion.js

```diff
@@ -0,0 +1,1851 @@
+ */
+var getOrCreateArchiveUser =  module.exports.getOrCreateArchiveUser = function(ctx, user, callback) {
+    
+    // Get and/or create archiveUser
+    PrincipalsDAO.getArchivedUser(user.tenant.alias, function(err, userArchive) {
+        if (err) {
+            return callback(err);
+
+        } else if (_.isEmpty(userArchive)) {
+            // Get email
+            var emailDomains = user.tenant.emailDomains;
+
+            // Persist the user object
+            var userOpts = {
+                'tenant': {'alias': user.tenant.alias},
+                'visibility': 'public',
+                'emailPreference': 'never',
+                'locale':  ctx.locale(),
+                'acceptedTC': null, 
+                'isUserArchive' : 'true'
+            };
+
+            var displayName = user.tenant.alias + ' archive';
+
+            var loginId = new LoginId(user.tenant.alias, AuthenticationConstants.providers.LOCAL, user.tenant.alias + 'Archive', {'password': 'password'});
+            // Create a record in the principals table
+            PrincipalsAPI.createUser(ctx, user.tenant.alias, displayName, userOpts, function(err, createdUser) {
+                if (err) {
+                    return callback(err);
+                }
+
+                // Create a user archive in the table archiveByTenant
+                PrincipalsDAO.createArchivedUser(user.tenant.alias, createdUser.id, function(err) {
+                    if (err) {
+                        return callback(err);
+                    } 
+
+                    // Get and return the userArchive
+                    PrincipalsDAO.getArchivedUser(user.tenant.alias, function(err, userArchive) {
+                        if (err) {
+                            return callback(err);
+                        }
+                        return callback(null, userArchive);
+                    });
+                });
+            });
+
+        } else {
+            return callback(null, userArchive);
+        }
+    });
+};    
+
+/**
+ * Delete rights on a user + update roles on editors && on user archive
+ *
+ * @param  {Context}    ctx                     Standard context object containing the current user and the current tenant
+ * @param  {String}     user                    User that will be deleted
+ * @param  {String}     archiveUser             User Archive
+ * @param  {Function}   callback                Standard callback function
+ * @param  {Object}     callback.err            An error that occured, if any
+ * @param  {Object}     callback.listEmail      Array of users to email
+ */
+var transferDataToArchive = module.exports.transferDataToArchive = function (ctx, user, archiveUser, callback) {
+    callback = callback || function() {};
+    var listEmail = [];
+    var listIdElement = [];
+
+    // Delete/update roles in contents
+    _transferRightToContent(ctx, user, archiveUser, listEmail, listIdElement, function(err, listEmail, listIdElement) {
```

Comment:
I've already thought about it... first of all, the content method aren't similar to the other methods because a content can have 3 different rights and not 2 as the other type of resource. 
Now, concerning folders, meetings, groups and discussions, I have to called a lot of methods based on the type of resource. I could do it but it will be a very long method with a lot of if condition ( if(variable.type === meeting) {MeetingAPI.getMeetingMembers(ctx, ...} ...). 
Maybe there is another easier way (?)

Reply:
It would likely need if- or switch-statements. It's something to think about, but if you can't think of a neater way to do it this will be OK too.